atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -16,123 +16,67 @@
 package com.palantir.atlasdb.sweep;
 
 import java.util.Optional;
-import java.util.OptionalLong;
 import java.util.Set;
-import java.util.concurrent.TimeUnit;
-
-import javax.annotation.Nullable;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
-import com.google.common.base.Stopwatch;
 import com.google.common.base.Supplier;
-import com.palantir.atlasdb.encoding.PtBytes;
 import com.palantir.atlasdb.keyvalue.api.InsufficientConsistencyException;
-import com.palantir.atlasdb.keyvalue.api.KeyValueService;
-import com.palantir.atlasdb.keyvalue.api.SweepResults;
 import com.palantir.atlasdb.keyvalue.api.TableReference;
-import com.palantir.atlasdb.schema.generated.SweepTableFactory;
-import com.palantir.atlasdb.sweep.priority.ImmutableUpdateSweepPriority;
 import com.palantir.atlasdb.sweep.priority.NextTableToSweepProvider;
 import com.palantir.atlasdb.sweep.priority.NextTableToSweepProviderImpl;
-import com.palantir.atlasdb.sweep.priority.SweepPriorityStore;
-import com.palantir.atlasdb.sweep.progress.ImmutableSweepProgress;
 import com.palantir.atlasdb.sweep.progress.SweepProgress;
-import com.palantir.atlasdb.sweep.progress.SweepProgressStore;
-import com.palantir.atlasdb.transaction.api.LockAwareTransactionManager;
 import com.palantir.atlasdb.transaction.api.Transaction;
-import com.palantir.atlasdb.transaction.api.TransactionManager;
 import com.palantir.atlasdb.transaction.api.TransactionTask;
-import com.palantir.atlasdb.transaction.impl.TxTask;
 import com.palantir.common.base.Throwables;
-import com.palantir.common.time.Clock;
 import com.palantir.lock.RemoteLockService;
 
 public final class BackgroundSweeperImpl implements BackgroundSweeper {
     private static final Logger log = LoggerFactory.getLogger(BackgroundSweeperImpl.class);
-    private final TransactionManager txManager;
     private final RemoteLockService lockService;
-    private final KeyValueService kvs;
-    private final SweepProgressStore sweepProgressStore;
-    private final SweepPriorityStore sweepPriorityStore;
     private final NextTableToSweepProvider nextTableToSweepProvider;
-    private final SweepTaskRunner sweepRunner;
     private final Supplier<Boolean> isSweepEnabled;
     private final Supplier<Long> sweepPauseMillis;
-    private final Supplier<SweepBatchConfig> sweepBatchConfig;
-    private final BackgroundSweeperPerformanceLogger sweepPerfLogger;
-    private final SweepMetrics sweepMetrics;
     private final PersistentLockManager persistentLockManager;
-    private final Clock wallClock;
+    private final SpecificTableSweeper specificTableSweeper;
 
-    private volatile double batchSizeMultiplier = 1.0;
+    static volatile double batchSizeMultiplier = 1.0;
```

Comment:
nit: I think this should remain private, it looks like it's only used here.

Also this prevents us from running multiple independent background sweepers in parallel, though we don't do that so we can leave that case to if/when we decide we might want that.